code/modules/projectiles/guns/energy/recharge.dm

```diff
@@ -0,0 +1,129 @@
+//Recharge subtype - used by stuff like protokinetic accelerators and ebows, one shot at a time, recharges.
+/obj/item/gun/energy/recharge
+	icon_state = "kineticgun"
+	base_icon_state = "kineticgun"
+	desc = "A self recharging gun. Holds one shot at a time."
+	automatic_charge_overlays = FALSE
+	cell_type = /obj/item/stock_parts/cell/emproof
+	/// If set to something, instead of an overlay, sets the icon_state directly.
+	var/no_charge_state
+	/// Does it hold charge when not put away?
+	var/holds_charge = FALSE
+	/// How much time we need to recharge
+	var/recharge_time = 1.6 SECONDS
+	/// Sound we use when recharged
+	var/recharge_sound = 'sound/weapons/kenetic_reload.ogg'
+	/// An ID for our recharging timer.
+	var/recharge_timerid
+	/// Do we recharge slower with more of our type?
+	var/unique_frequency = FALSE
+
+/obj/item/gun/energy/recharge/Initialize(mapload)
+	. = ..()
+	if(!holds_charge)
+		empty()
+
+/obj/item/gun/energy/recharge/shoot_live_shot(mob/living/user, pointblank = 0, atom/pbtarget = null, message = 1)
+	. = ..()
+	attempt_reload()
+
+/obj/item/gun/energy/recharge/equipped(mob/user)
+	. = ..()
+	if(!can_shoot())
+		attempt_reload()
+
+/obj/item/gun/energy/recharge/dropped()
+	. = ..()
+	if(!QDELING(src) && !holds_charge)
+		// Put it on a delay because moving item from slot to hand
+		// calls dropped().
+		addtimer(CALLBACK(src, .proc/empty_if_not_held), 0.1 SECONDS)
+
+/obj/item/gun/energy/recharge/handle_chamber()
+	. = ..()
+	attempt_reload()
+
+/obj/item/gun/energy/recharge/proc/empty_if_not_held()
+	if(!ismob(loc))
+		empty()
+		deltimer(recharge_timerid)
+
+/obj/item/gun/energy/recharge/proc/empty()
+	if(cell)
+		cell.use(cell.charge)
+	update_appearance()
+
+/obj/item/gun/energy/recharge/proc/attempt_reload(set_recharge_time)
+	if(!cell)
+		return
+	if(cell.charge == cell.maxcharge)
+		return
+	if(!set_recharge_time)
+		set_recharge_time = recharge_time
+	var/carried = 0
+	if(!unique_frequency)
+		for(var/obj/item/gun/energy/recharge/recharging_gun in loc.get_all_contents())
+			if(recharging_gun.type != type || recharging_gun.unique_frequency)
+				continue
+			carried++
+		carried = max(carried, 1)
```

Comment:
Unindent this or anything anything without unique frequency will instantly recharge